docs/src/man/doctests.md

```diff
@@ -105,59 +102,102 @@ line where checking should stop, i.e.
 julia> div(1, 0)
 ERROR: DivideError: integer division error
 [...]
+```
+````
+
+## Preserving definitions between blocks
+
+Every doctest block is evaluated inside its own `module`. This means that definitions
+(types, variables, functions etc.) from a block can *not* be used in the next block.
+A doctest block can be labeled in order to collect several blocks into the same module.
+All blocks with the same label (in the same file) will be evaluated in the same module,
+and hence share scope. This can be useful if the same definitions are used in more than
+one block, with for example text, or other doctest blocks, in between. Example:
+
+````markdown
```

Comment:
As I was reading through the text, the illustrative example felt a bit long, so I personally might prefer having it be a bit more concise and divided up.

I would propose the following alternative: have something along these lines after the first two sentences to illustrate the problem:

````
```jldoctest
julia> foo = 42
42
```

```jldoctest
julia> println(foo)
ERROR: UndefVarError: foo not defined
```
````

And then show how you can solve the issue with labels:
````
```jldoctest mylabel
julia> foo = 42
42
```

```jldoctest mylabel
julia> println(foo)
42
```
````

I would also move all the explanatory text outside of the example.